app/controllers/bookings_controller.rb

```diff
@@ -0,0 +1,40 @@
+class BookingsController < ApplicationController
+  before_action :find, only: %i[show destroy]
+  before_action :find_dream, only: %i[new create]
+
+  def show; end
+
+  def new
+    @booking = Booking.new
+  end
+
+  def create
+    @booking = Booking.new(booking_params)
+    @booking.user = current_user
+    @booking.dream = @dream
+    if @booking.save
+      redirect_to dream_path(@booking.dream)
+    else
+      render :new
+    end
+  end
+
+  def destroy
+    @booking.destroy
+    redirect_to root
```

Comment:
to give the right status I would change this line in:

"redirect_to root_path, status: :see_other"